lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestResubmitTask.cs

```diff
@@ -0,0 +1,263 @@
+            private int _taskNumber = 1;
+            private const string ContextId = "ContextId";
+            private int _contextNumber = 1;
+            private readonly IDictionary<string, string> _contextTaskMapping = new Dictionary<string, string>();
+            private readonly object _lock = new object();
+
+            [Inject]
+            private ResubmitTaskTestDriver(IEvaluatorRequestor evaluatorRequestor)
+            {
+                _requestor = evaluatorRequestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().SetNumber(2).Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                lock (_lock)
+                {
+                    value.SubmitContext(
+                        ContextConfiguration.ConfigurationModule
+                            .Set(ContextConfiguration.Identifier, ContextId + _contextNumber)
+                            .Build());
+                    _contextNumber++;
+                }
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                SubmitTask(value);
+            }
+
+            private void SubmitTask(IActiveContext context)
+            {
+                lock (_lock)
+                {
+                    context.SubmitTask(GetTaskConfigurationForCloseTask(TaskId + _taskNumber));
+                    _contextTaskMapping.Add(TaskId + _taskNumber, context.Id);
+                    _taskNumber++;
+                }
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                Logger.Log(Level.Info, "Task completed: " + value.Id);
+                VerifyContextTaskMapping(value.Id, value.ActiveContext.Id);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(IFailedTask value)
+            {
+                var failedExeption = ByteUtilities.ByteArraysToString(value.Data.Value);
+                Logger.Log(Level.Error, "In IFailedTask: " + failedExeption);
+
+                VerifyContextTaskMapping(value.Id, value.GetActiveContext().Value.Id);
+                Assert.Contains(TaskKilledByDriver, failedExeption);
+
+                SubmitTask(value.GetActiveContext().Value);
+            }
+
+            private void VerifyContextTaskMapping(string taskId, string contextId)
+            {
+                lock (_lock)
+                {
+                    string expectedContextId;
+                    _contextTaskMapping.TryGetValue(taskId, out expectedContextId);
+                    Assert.Equal(expectedContextId, contextId);
+                }
+            }
+
+            public void OnNext(IRunningTask value)
+            {
+                Logger.Log(Level.Info, "Task running: " + value.Id);
+                switch (value.Id)
+                {
+                    case TaskId + "1":
```

Comment:
TaskId is already a constant. "1" is only used here, it doesn't add value to define it as a constant.